File: src/main/java/com/github/shatteredsuite/scrolls/data/scroll/NBTReader.java
package com.github.shatteredsuite.scrolls.data.scroll;

import com.github.shatteredsuite.core.include.nbt.NBTCompound;
import com.github.shatteredsuite.core.include.nbt.NBTItem;

/**
 * General functionality to create BindingData from NBT.
 */
public interface NBTReader {

    /**
     * Creates BindingData from an NBT item.
     * The inverse of {@link NBTApplier#applyNBT(NBTItem)}.
     *
     * @param compound The compound to read data from to create BindingData.
     * @return The BindingData created from the item.
     */
    BindingData fromNBT(NBTCompound compound);
}

File: src/main/kotlin/com/github/shatteredsuite/scrolls/data/scroll/NBTReader.kt
package com.github.shatteredsuite.scrolls.data.scroll

import com.github.shatteredsuite.core.include.nbt.NBTCompound
import com.github.shatteredsuite.scrolls.data.scroll.binding.BindingData

/**
 * General functionality to create BindingData from NBT.
 */
interface NBTReader {
    /**
     * Creates BindingData from an NBT item.
     * The inverse of [NBTApplier.applyNBT].
     *
     * @param compound The compound to read data from to create BindingData.
     * @return The BindingData created from the item.
     */
    fun fromNBT(compound: NBTCompound): BindingData
}